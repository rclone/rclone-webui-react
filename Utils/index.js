import checkPropTypes from 'check-prop-types';
import {applyMiddleware, createStore} from 'redux';
import rootReducer from './../src/reducers';
import {middleware} from '../src/store';

export const findByTestAttr = (component, attr) => {
    const wrapper = component.find(`[data-test='${attr}']`);
    return wrapper;
};

export const checkProps = (component, expectedProps) => {
    const propsErr = checkPropTypes(component.propTypes, expectedProps, 'prop', component.name);
    return propsErr;
};

export const testStore = (initialState) => {
    const createStoreWithMiddleware = applyMiddleware(...middleware)(createStore);
    return createStoreWithMiddleware(rootReducer, initialState);
};