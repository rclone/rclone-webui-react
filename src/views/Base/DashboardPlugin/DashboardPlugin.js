import React from "react";
import Iframe from "react-iframe";
import * as PropTypes from "prop-types";

/**
 * Dashboard plugin loads a plugin in an iframe
 * @param pluginUrl {string}
 * @return {*}
 * @constructor
 */
function DashboardPlugin({pluginUrl}) {
    return (
        <div data-test="dashboardPluginComponent">
            {(pluginUrl) ? <Iframe url={pluginUrl}
                                   style={{border: "none"}}
                                   allowTransparency="true"
                                   allowFullScreen="true"
                                   width="100%"
                                   height="100%"
                                   display="initial"/>
                : <p>Could not find any appropriate plugin to load</p>}
        </div>
    );
}

DashboardPlugin.propTypes = {
    pluginUrl: PropTypes.string.isRequired
}

export default DashboardPlugin;