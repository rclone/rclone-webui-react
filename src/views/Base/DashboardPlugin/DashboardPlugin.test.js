import React from "react";
import {shallow} from "enzyme";
import toJson from "enzyme-to-json";
import DashboardPlugin from "./DashboardPlugin";
import {findByTestAttr, testStore} from "../../../../Utils";


const setUp = (intialState = {}, props = {}) => {
    const store = testStore(intialState);

    const component = shallow(<DashboardPlugin {...props} store={store}/>);
    return component;
};


describe('Dashboard Plugin', function () {

    describe('renders', function () {
        let wrapper;
        beforeEach(() => {
            const initialState = {};

            const props = {
                pluginUrl: ""
            };
            wrapper = setUp(initialState, props)
        });

        it('should render without crashing', function () {
            const component = findByTestAttr(wrapper, "dashboardPluginComponent");
            expect(component).toHaveLength(1);
        });

        it('should match snapshot', function () {
            expect(toJson(wrapper)).toMatchSnapshot()
        });
    });

});
