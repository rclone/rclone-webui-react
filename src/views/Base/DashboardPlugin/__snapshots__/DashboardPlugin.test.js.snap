// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Plugin renders should match snapshot 1`] = `
<div
  data-test="dashboardPluginComponent"
>
  <p>
    Could not find any appropriate plugin to load
  </p>
</div>
`;
